[ajjajjajjajj] iP

ajjajjajjajj opened the pull request:
# DukeProPro

> “Why waste time say lot word when few word do trick” -Kevin Malone

DukePro frees your mind of having to say lot word for thing you need to do. It's

- text-based
- easy to learn
- ~FAST~ SUPER FAST to use
- more pro than pro la

All you need to do is,

1. download it from [here](https://github.com/ajjajjajjajj/ip.git).
2. double-click it.
3. add your tasks.
4. let it manage your tasks for you 😉

And it is **FREE**!

Features:

 

- [x] Managing tasks
- [ ]  Managing deadlines (coming soon)
- [ ]  Reminders (coming soon)

---

If you Java programmer, you can use it to practice Java too. Here's the main method:

```
public class Main {
    public static void main(String[] args) {
        Application.launch(MainApp.class, args);
    }
}
```

Commit:
Add Gradle support

seanleong339 committed:
.gitignore: Fix ACTUAL.txt -> ACTUAL.TXT  (#73)

text-ui-test/runtest.bat|sh scripts generate a file ACTUAL.TXT.
However, .gitignore uses ACTUAL.txt, which means the generated
file will not be ignored by Git on non-Windows OS.

Let's update .gitignore as ACTUAL.txt -> ACTUAL.TXT

ajjajjajjajj committed:
added gitignore

ajjajjajjajj committed:
added gitignore

ajjajjajjajj committed:
added skeletal version

ajjajjajjajj committed:
added skeletal version

ajjajjajjajj committed:
added list storage function

ajjajjajjajj committed:
added list storage function

ajjajjajjajj committed:
added marking/unmarking feature

ajjajjajjajj committed:
added marking/unmarking feature

ajjajjajjajj committed:
added support for different tasks

ajjajjajjajj committed:
added support for different tasks

ajjajjajjajj committed:
added exception handling for adding tasks

ajjajjajjajj committed:
added exception handling for adding tasks

ajjajjajjajj committed:
added delete function

ajjajjajjajj committed:
added delete function

ajjajjajjajj committed:
Add text-UI testing

ajjajjajjajj committed:
Add text-UI testing

ajjajjajjajj committed:
Add Storage class and basic support for loading from Storage

ajjajjajjajj committed:
Add Storage class and basic support for loading from Storage

ajjajjajjajj committed:
Rework classes to add skeletal TaskList, Parser, Ui, Storage and Command

ajjajjajjajj committed:
Rework classes to add skeletal TaskList, Parser, Ui, Storage and Command

ajjajjajjajj committed:
Restored functionality based on new class structure

ajjajjajjajj committed:
Restored functionality based on new class structure

ajjajjajjajj committed:
Add functionality and support for Storage class - save and load

ajjajjajjajj committed:
Add functionality and support for Storage class - save and load

ajjajjajjajj committed:
Implement datetime formats for Deadline and Event tasks

ajjajjajjajj committed:
Implement datetime formats for Deadline and Event tasks

ajjajjajjajj committed:
Resolve merge conflicts for LocalDateTime integration

ajjajjajjajj committed:
Organise classes into packages

ajjajjajjajj committed:
Organise classes into packages

ajjajjajjajj committed:
Update package organisation

ajjajjajjajj committed:
Update package organisation

ajjajjajjajj committed:
Revert "Update package organisation"

This reverts commit 9bb4b168f444496d78afdccd6bccba6e269bcb9d.

ajjajjajjajj committed:
Merge branch 'add-gradle-support'

ajjajjajjajj committed:
Add support for gradl, including checkstyle and unit tests

ajjajjajjajj committed:
Add compulsory files for Gradle checkstyle

ajjajjajjajj committed:
Modify build.gradle

ajjajjajjajj committed:
Establish sample storage data for pre-loading

ajjajjajjajj committed:
Modify class packages

ajjajjajjajj committed:
Add test classes and test requirements

ajjajjajjajj committed:
Edit test classes

ajjajjajjajj committed:
Refactor TaskTypes to change from class to enum

ajjajjajjajj committed:
Add support for Find function

ajjajjajjajj committed:
Modify code to comply with Java coding standards

ajjajjajjajj committed:
Add JavaDoc for non-trivial methods and classes

ajjajjajjajj committed:
Resolve merge conflict

ajjajjajjajj committed:
Resolve merge conflict

ajjajjajjajj committed:
lol

ajjajjajjajj committed:
lol2

ajjajjajjajj committed:
lol3

ajjajjajjajj committed:
Add JavaFX

ajjajjajjajj committed:
Add basic GUI - yet to integrate Duke into GUI

ajjajjajjajj committed:
Fix GUI bugs

ajjajjajjajj committed:
Integrate functional GUI for chatbot

ajjajjajjajj committed:
Update README.md

WillCWX commented on src/main/java/app/chatbot/Duke.java:
```diff
@@ -0,0 +1,92 @@
+import app.command.SaveCommand;
+import app.task.TaskList;
+
+public class Duke {
```
> You MUST write header comments for all classes, public methods.

The JavaDoc for the public class Duke is missing. Perhaps a one-liner would do?

ajjajjajjajj replied:
> Well done on the GUI, it looks amazing! The different packages and class show good OOP paradigms and effort on your iP.
> 
> There are only minor JavaDoc and style issues. Mainly `white spaces` and `unused variables`. The missing JavaDocs for the methods can all be one-liners so they are not egregious at all. I have highlighted some of these trivial errors but seeing as how they repeat, I have attached the [Full Checkstyle report here](https://github.com/nus-cs2103-AY2223S2/ip/files/10607890/main.2.docx) instead of commenting one by one.

Much appreciated for the feedback! Concur on the issues raised as well - will address them in the next commit.
The note on the BufferReader was helpful and prompted some extra learning at my end as well.

WillCWX commented on src/main/java/app/chatbot/Duke.java:
```diff
@@ -0,0 +1,92 @@
+    private Ui ui;
+
+
+    public Duke() {
```
Running gradle check shows that this statement requires a JavaDoc but I understand that this constructor is pretty self-explanatory.

Perhaps a one-liner would do as well?

WillCWX commented on src/main/java/app/chatbot/Duke.java:
```diff
@@ -0,0 +1,92 @@
+            }
+            return response.toString();
+        } catch (Exception e) {
+            return(e.getMessage());
```
> Java reserved words should be followed by a white space.

According to [cs2103t->layout->white space](https://se-education.org/guides/conventions/java/index.html#layout), there needs to be a space between `return` and `(e.getMessage())`
This also occurs below at `loadStorageData`.

WillCWX commented on src/main/java/app/chatbot/Duke.java:
```diff
@@ -0,0 +1,92 @@
+        }
+    }
+
+    public String loadStorageData() {
```
Missing JavaDocs

WillCWX commented on src/main/java/app/chatbot/Duke.java:
```diff
@@ -0,0 +1,92 @@
+    public Duke() {
+        this.ui = new Ui();
+        try {
+            this.storage = new Storage(this.STORAGE_LOCATION);
```
`STORAGE_LOCATION` is a static variable and should not be accessed using the Object variable identifier `this`. It should be accessed using the class name `Duke` instead or just simply accessed without any identifier, `STORAGE_LOCATION`.
```suggestion
            this.storage = new Storage(Duke.STORAGE_LOCATION);
```

WillCWX commented on src/main/java/app/chatbot/Duke.java:
```diff
@@ -0,0 +1,92 @@
+
+
+    public String getResponse(String fullCommand) {
+        boolean isExit = false;
```
This variable appears to be unused. Is this needed for something?

WillCWX commented on src/main/java/app/chatbot/Response.java:
```diff
@@ -0,0 +1,27 @@
+        this.sb = new StringBuilder();
+    }
+
+    public Response(String s) {
```
Missing JavaDocs, one-liners should do. 

The same goes for the method below (Without the JavaDoc for `addLine` users don't know that the line is added to the stringbuilder rather than the input string itself)
```
public void addLine(String input) {
```

WillCWX commented on src/main/java/app/chatbot/Storage.java:
```diff
@@ -0,0 +1,135 @@
+        return 1;
+    }
+
+    public static class InvalidStorageException extends Exception {
```
Another one-liner JavaDoc?

WillCWX commented on src/main/java/app/chatbot/Storage.java:
```diff
@@ -0,0 +1,135 @@
+    public static final String SEPARATOR_REGEX = " \\| ";
+    public static final String SEPARATOR = " | ";
+    public static final String SUB_SEPARATOR = ":";
+    private Path path;
```
This variable is not used. Is it needed somewhere?

WillCWX commented on src/main/java/app/chatbot/Storage.java:
```diff
@@ -0,0 +1,135 @@
+        try {
+            BufferedReader br = new BufferedReader(new FileReader(this.file));
+            String line = br.readLine();
+            int totalTaskCounter = 0;
```
This variable is not used as well. DId you forget to delete it?

WillCWX commented on src/main/java/app/chatbot/Storage.java:
```diff
@@ -0,0 +1,135 @@
+
+
+        try {
+            BufferedReader br = new BufferedReader(new FileReader(this.file));
```
The `BufferedReader br` is not closed and can cause a resource leak. Maybe you can add `br.close()` right before you `return totalSuccess`?

WillCWX reviewed (changes requested):
Well done on the GUI, it looks amazing! The different packages and class show good OOP paradigms and effort on your iP. 

There are only minor JavaDoc and style issues. Mainly `white spaces` and `unused variables`. The missing JavaDocs for the methods can all be one-liners so they are not egregious at all. 
I have highlighted some of these trivial errors but seeing as how they repeat, I have attached the [Full Checkstyle report here](https://github.com/nus-cs2103-AY2223S2/ip/files/10607890/main.2.docx) instead of commenting one by one.

ajjajjajjajj committed:
Resolve PR issues and minor refactoring

arkarsg commented on src/main/java/app/chatbot/Duke.java:
```diff
@@ -0,0 +1,92 @@
+package app.chatbot;
```
Hi. I am Arkar and I will be reviewing the coding standard.

ajjajjajjajj replied:
Thanks for the in-depth review! The changes you suggested are really sound - I'll be tackling them in future commits for this week.

arkarsg commented on src/main/java/app/chatbot/Duke.java:
```diff
@@ -0,0 +1,92 @@
+            return(e.getMessage());
+        }
+    }
+}
```
No issues here for coding standard. A very clean and readable implementation of what was suggested in the notes. Nice!

arkarsg commented on src/main/java/app/chatbot/Parser.java:
```diff
@@ -0,0 +1,99 @@
+import java.util.List;
+import java.util.Map;
+
+import app.command.AddCommand;
```
Good ordering and grouping of import statements. Good use of class names as well.

arkarsg commented on src/main/java/app/chatbot/Parser.java:
```diff
@@ -0,0 +1,99 @@
+        String[] args = input.split(" /");
+
+        // process main command
+        String[] mainCommand = args[0].split(" ", 2);
```
I like how this is handled instead of tokenising every word of input to give [event, attend, wedding]

arkarsg commented on src/main/java/app/chatbot/Parser.java:
```diff
@@ -0,0 +1,99 @@
+        // raw format according to task is /<name> <value>
+
+        Map<String, String> map = new HashMap<>();
+        String[] args = input.split(" /");
```
Not really an issue to use `args` here but be careful if this was in the main class.

arkarsg commented on src/main/java/app/chatbot/Response.java:
```diff
@@ -0,0 +1,27 @@
+package app.chatbot;
+
+/**
+ * Helper class for formatting GUI responses.
```
Nice addition

arkarsg commented on src/main/java/app/chatbot/Storage.java:
```diff
@@ -0,0 +1,135 @@
+     * @param tl empty TaskList, assuming this is executed upon startup of app.
+     * @return totalSuccess, the number of lines that have been successfully loaded into the tl.
+     */
+    public int loadIntoTaskList(TaskList tl) throws Exception {
```
Is it possible to narrow which `Exception` is thrown? From the looks of it, there can be `EOFException`, `IOException` and `FileNotFoundException`. I think this helps for JavaDocs too.

ajjajjajjajj replied:
Indeed, this is a helpful suggestion. I'll be implementing better exception handling for this week!

arkarsg commented on src/main/java/app/chatbot/Storage.java:
```diff
@@ -0,0 +1,135 @@
+        return 1;
+    }
+
+    public static class InvalidStorageException extends Exception {
```
Alternative: exception as a class of its own.

ajjajjajjajj replied:
Noted on this, will do some reading up on best practices for extending Exception classes.

arkarsg commented on src/main/java/app/chatbot/Ui.java:
```diff
@@ -0,0 +1,53 @@
+        return sc.nextLine();
+    }
+
+    public void showWelcome() {
```
Not sure if this is the latest commit but some public methods are missing JavaDocs while some are present.

arkarsg commented on src/main/java/app/command/FindCommand.java:
```diff
@@ -0,0 +1,38 @@
+            }
+        }
+        if (!isSearchSuccessful) {
+            return ("Ah I didn't find any tasks matching '" + this.searchContent + "'." + "\n" );
```
Nice user design that you show the search term back to the user

arkarsg reviewed (commented):
Overall, looks great! Would recommend against `throws Exception`

ajjajjajjajj committed:
Add assertion checks

ajjajjajjajj committed:
Modify gitignore

ajjajjajjajj committed:
Modify gitignore

ajjajjajjajj committed:
Edited gradle configs

ajjajjajjajj committed:
Refactor storage-related code and added debug messages

ajjajjajjajj committed:
Refactor code

ajjajjajjajj committed:
Merge pull request #2 from ajjajjajjajj/branch-A-CodeQuality

Merge branch-A-CodeQuality

ajjajjajjajj committed:
Merge with master

ajjajjajjajj committed:
Merge pull request #1 from ajjajjajjajj/branch-A-Assertions

Merge Branch-A-Assertions

ajjajjajjajj committed:
Fix coding style issues

After adding in new features for week 5, several violations
of coding style conventions have appeared. This is a very serious issue
because the reader can get confused.

Read the coding styles. Run checkstyle. Fix coding violations as
they're blasphemic.

ajjajjajjajj committed:
Update README

ajjajjajjajj committed:
Add new Edit Command

Allows the user to indicate which Task to edit. The number of fields
required is at least 1, with the consideration that the user might
only want to change a subset of attributes.

To be more specific, after an edit, a new Task is created and replaces
the pre-existing (to be edited) task in the TaskList. This was
implemented as so to maintain some immutability of Tasks.

ajjajjajjajj committed:
Refactor code and move class methods

Error messages were not in each classes' static variables. To
comply with coding standards and have greater readability,
error messages and reply statements have been moved to static
final attributes in each class.

Previously, each executed Command returns a simple string. With
consideration for the need for greater detail in a response,
a flag isSuccess has been added. The flag serves as an indicator
if an executed Command should have changes be written in. This
eliminates the risk of a failed Command execution resulting in results
being saved.

ajjajjajjajj committed:
Set up minimal CI

ajjajjajjajj committed:
Update CI

ajjajjajjajj committed:
Add Summary function

The summary function invokes whenever the app is booted up and
storage has been loaded. It provides an overview of the tasks that
the user has, and is sorted in a neat fashion for easy reading.

A view of the summary can be invoked at anytime by the user through
the 'summary' command.

However, it is to be noted that the summary command does not give the
user any information that allows them to access a certain Task (to
modify, mark, or delete). This might cause some confusion, but
is mitigated by the closing message at the end of a summary.

ajjajjajjajj committed:
Update documentation

ajjajjajjajj committed:
Fix text output bugs

ajjajjajjajj committed:
Create _config.yml

ajjajjajjajj committed:
Update _config.yml

ajjajjajjajj committed:
Update _config.yml

ajjajjajjajj committed:
Add default tasks loading and jar bundling for
storage

Previously, the storage file was placed in the root directory of the
project. This resulted in the storage not being bundled in the JAR file.

Storage has now been moved into resources. In addition, a default tasks
storage has been added to load in some default, initial tasks when the
app is first launched.

ajjajjajjajj committed:
Fix styling issues

ajjajjajjajj committed:
Added a welcome message function for first-time bootup

ajjajjajjajj committed:
Delete _config.yml

ajjajjajjajj committed:
Update README.md
